### PR TITLE
Jetty 12 - Javadoc cleanup & new inspection profile

### PR DESCRIPTION
**Pull request by joakime**
Introduce importable intellij inspection profile for Javadoc.

Perform some javadoc cleanup as well.

### CODE REVIEW

**Comment by joakime** [comment]
The inspection profile for just javadoc, suitable for import, can be found in `build/build-resources/intellij-javadoc-inspection.xml`

It has more inspections than just the javadoc doclint does.

For example, here's an interesting one for "Mismatch between Javadoc and code" ...

intellij-javadoc-inspection-profile

The javadoc talks about a return type as if it's a boolean, but the return type is anything but.

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/security/UnixCrypt.java [hunk 1]
```diff
@@ -28,11 +28,6 @@
  * by Iris Van den Broeke, Daniel Deville
  */
 
-/**
- * modified April 2001
- * by Iris Van den Broeke, Daniel Deville
- */
-
```
Comment (joakime) [inline jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/security/UnixCrypt.java:35]: This exact block is a duplicate of what is already present a few lines above.

File: jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/internal/WebSocketSessionState.java [hunk 2]
```diff
@@ -129,7 +128,7 @@ public boolean onClosed(CloseStatus closeStatus)
      * </p>
      * <p>
      * This should only be called if there is an error directly before the call to
-     * {@link WebSocketCoreSession#closeConnection(CloseStatus, Callback)}.
+     * {@code WebSocketCoreSession.closeConnection(CloseStatus, Callback)}.
```
Comment (joakime) [inline jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/internal/WebSocketSessionState.java:131]: the `.closeConnection(CloseStatus, Callback)` is not accessible from here and cannot be a javadoc link.

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/FileID.java [hunk 3]
```diff
@@ -142,14 +142,14 @@ public static String getExtension(URI uri)
...
-     * <code>{@code
+     * <pre>{@code
      * "foo.tar.gz" "gz"
      * "foo.bar"    "bar"
      * "foo."       ""
      * "foo"        null
      * ".bar"       null
      * null         null
-     * }</code>
+     * }</pre>
```
Comment (joakime) [inline jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/FileID.java:152]: This preserves the LF and whitespace

**Comment by joakime** [comment]
@gregw this good to merge? (if so, go ahead and do it on your timezone)

**Review by gregw — APPROVED** [review]
This is good on it's own, but we need to do more to document how to use this (and the style guides and the check style).  Perhaps `CONTRIBUTING.md` should have some instructions.

Perhaps @gregpoulos you can put this on your list of documentations?   Actually, working of fixing some of the javadoc issues would also be a good way for you to get a tour of the code.

**Comment by gregpoulos** [comment]
@gregw Yep, I'll add this to my list of things to look at!

**Pull request merged**